docs/reference/filebeat/delete-file-guide.md

```diff
@@ -0,0 +1,140 @@
+
+If any of the checks fail, the harvester is closed. One the next
+file system scan happens, a new harvester, and reader, will be
+started, once the close condition (EOF or inactivity) is met, then the
+remove process will start again.
+
+Once all checks are successful the file is removed.
+
+## EOF x Inactivity
+Filestream's reader can be configured to close on two conditions: EOF
+and inactivity, each one has a different purpose:
+ - EOF: it is recommended for files that do not have data appended to
+   them, like a cronjob that when it is done copies the file to a
+   folder where Filestream can read it;
+ - Inactive: it is recommended for files that have data appended to
+   them, like long running process that does not performs its own log
+   rotation.
+ 
+::::{note}
+Even for short lived process that write their own log file within a
+few seconds, avoid using EOF because Filestream might read until EOF
+before the last entries are written to the file.
+::::
+
+## Examples
+### Log files from old cronjobs
+Filebeat will be used to ingest log files from old cronjobs, all files
+have been fully written and Filebeat should remove them once it
+finishes publishing all data. The log files are located at
+`/var/log/cronjobs/*.log`.
+
+For that the Filestream with delete on EOF will be used, the input
+configuration is:
+```yaml
+  - type: filestream
+    id: cronjobs-logs
+    paths:
+      - /var/log/cronjobs/*.log
+    delete.on_close.eof: true
+    delete.grace_period: 0s
+```
+
+### Log files from long running tasks
+Filebeat will be used to ingest log files from tasks that run for a
+few minutes, appending to their log files, if the tasks are still
+running log entries are added every few seconds. Once no more entries
+have been added for a few minutes, it is safe to remove the log file. The
+log files are located at `/var/log/long-tasks/*.log`.
+
+For this case Filestream can be configured to remove files after a
+period of inactivity, the simplest configuration is:
+
+```yaml
+  - type: filestream
+    id: long-tasks-logs
+    paths:
+      - /var/log/long-tasks/*.log
+    delete.on_close.inactive: true
+    delete.grace_period: 0s
+```
+
+Because `delete.on_close.inactive: true` the time to consider a file
+inactive and close the reader is automatically set to 30 minutes. This
+can be overwritten to a short or longer time, e.g: 5 minutes.
+
+```yaml
+  - type: filestream
+    id: long-tasks-logs
+    paths:
+      - /var/log/long-tasks/*.log
+    delete.on_close.inactive: true
+    close.on_state_change.inactive: 5m
+    delete.grace_period: 0s
+```
+
+### Waiting before removing log files
+It is also possible to configure a grace period to wait after the
+all events have been published before removing the file. Note that
+this is different than the 'close on inactive' because the inactivity
+timeout for the reader does not consider if an event has been
+published, this means a file can be closed due to inactivity (no more
```

Comment:
Giving the grace period a different semantic meaning than the inactivity timer seems confusing / complicated. If we've fully read a file and it's been idle for 12 hours while the ES connection was down, and the connection comes back up and we finish ingesting all of the file's data, and it's still idle, why would we still want to wait an extra grace period instead of deleting immediately? Don't we just want to wait until both the file is fully ingested and it has been idle/closed for the configured grace period? The grace period should be configured based on file access patterns, so it doesn't seem like it should be affected by how fast ingestion is proceeding, and this interaction makes the configuration harder to explain.

Reply:
> Don't we just want to wait until both the file is fully ingested and it has been idle/closed for the configured grace period? The grace period should be configured based on file access patterns, so it doesn't seem like it should be affected by how fast ingestion is proceeding (...)

That's the role of close on inactive.

I'm happy to discuss more about it, I'm not a fan of this grace period. Maybe my arguments at the time were not good enough.